### ISSUE TITLE
6D closed orbit not added to particles tracked in 4D

### ISSUE THREAD

**Issue by rdemaria** [SixTrack/SixTrack#337]
When 4D tracking is chosen, iclo6 should be 1 (4D closed orbit) instead of 2 (6D closed orbit), otherwise 6D closed orbit is calculated but not added to the initial conditions.

The fix should be that 6D closed orbit is added to the initial conditions.

**Comment by kyrsjo** [comment]
The alternative fix is that we disable mixing of 6D / 4D tracking and optics?

Note that the bug is probably relevant for PR #316 .

**Comment by rdemaria** [comment]
Actually for @JamesMolson, the iclo6=5 or 6 is reported can use a guess for the closed orbit search.

The documentation is also not very clear, perhaps we should review what the option does. I am afraid of uncontrolled side effects...

**Comment by kyrsjo** [comment]
Also, if the file with the guess is not read with the support for correct rounding, we may end up with different results on different hosts on BOINC... Should be checked!

**Comment by kyrsjo** [comment]
Looking through this now - according to the manual, the meaning of iclo6 is:
* 0: 6D closed orbit not calculated
* 1: 6D closed orbit calculated using the DA package but not added to the initial coordinates
* 2: 6D closed orbit calculated using the DA package and added to the initial coordinates
* 5/6: Like 1 and 2, however additionally a guess for the closed orbit is read in from fort.33.

Furthermore, there is the flag idfor, which has the meaning:
* 0: closed orbit added
* 1: Inital coordinates unchanged
* 2: Read distribution from fort.13

Noe that
```
      if(iclo6.eq.2.and.idfor.eq.0) idfor=1
```
and the coordinates are (in `maincr`) added as follows
```
          if(iclo6.eq.2) then
            xv(1,ia)=xv(1,ia)+clo6v(1,ia)
            yv(1,ia)=yv(1,ia)+clop6v(1,ia)
            xv(2,ia)=xv(2,ia)+clo6v(2,ia)
            yv(2,ia)=yv(2,ia)+clop6v(2,ia)
            sigmv(ia)=sigmv(ia)+clo6v(3,ia)
            dpsv(ia)=dpsv(ia)+clop6v(3,ia)
            xv(1,ia+1)=xv(1,ia+1)+clo6v(1,ia)
            yv(1,ia+1)=yv(1,ia+1)+clop6v(1,ia)
            xv(2,ia+1)=xv(2,ia+1)+clo6v(2,ia)
            yv(2,ia+1)=yv(2,ia+1)+clop6v(2,ia)
            sigmv(ia+1)=sigmv(ia+1)+clo6v(3,ia)
            dpsv(ia+1)=dpsv(ia+1)+clop6v(3,ia)
            oidpsv(ia)=one/(one+dpsv(ia))
            oidpsv(ia+1)=one/(one+dpsv(ia+1))
          endif
          if(iclo6.ne.2) then
            xv(1,ia)=xv(1,ia)+(clov(1,ia)*dble(idz(1)))*dble(1-idfor)    !hr05
            yv(1,ia)=yv(1,ia)+(clopv(1,ia)*dble(idz(1)))*dble(1-idfor)   !hr05
            xv(2,ia)=xv(2,ia)+(clov(2,ia)*dble(idz(2)))*dble(1-idfor)    !hr05
            yv(2,ia)=yv(2,ia)+(clopv(2,ia)*dble(idz(2)))*dble(1-idfor)   !hr05
          xv(1,ia+1)=xv(1,ia+1)+(clov(1,ia)*dble(idz(1)))*dble(1-idfor)  !hr05
          yv(1,ia+1)=yv(1,ia+1)+(clopv(1,ia)*dble(idz(1)))*dble(1-idfor) !hr05
          xv(2,ia+1)=xv(2,ia+1)+(clov(2,ia)*dble(idz(2)))*dble(1-idfor)  !hr05
          yv(2,ia+1)=yv(2,ia+1)+(clopv(2,ia)*dble(idz(2)))*dble(1-idfor) !hr05
          endif
```
I.e. the idfor=0 is only used to add 4d closed orbit; it can be overridden by setting iclo6=2 which will causes it to always use the 6D calculation (which AFAIK will not work very well for 4D optics...).

**Comment by kyrsjo** [comment]
After looking at various inputs and outputs (essentially the example files in #316 ), I've convinced myself that having clo6d not switched on when doing 6d tracking is just wrong - it leads to a 4x4 TAS matrix (matrix of eigenvectors, used for going from normalized to non-normalized coordinates) being computed and used for the initial particle positions. As an example:
![screenshot from 2017-11-05 23-24-30](https://user-images.githubusercontent.com/1011920/32419817-92eb0a1e-c280-11e7-8d15-32eda2c4de51.png)
![screenshot from 2017-11-05 23-24-43](https://user-images.githubusercontent.com/1011920/32419818-9524c112-c280-11e7-9aa9-150245d74d86.png)
Left: iclo6=0, Right: iclo6=1

I will therefore add a check when starting 4d- or 6d tracking, saying that if you're doing 6D tracking then assert iclo6.ne.0 (or ERROR); conversely if 6d then assert iclo6.eq.0.

**Comment by kyrsjo** [comment]
Fixed in PR #316
